### PR TITLE
feat(server,dashboard): advertise skill_trust_accept via auth_ok capabilities (#3272)

### PR DESCRIPTION
**Pull request by blamechris**
## Summary

- Server advertises a capability map in \`auth_ok\` (\`capabilities: { skillTrustAccept: true }\`).
- Protocol schema gains an optional \`capabilities\` field on \`ServerAuthOkSchema\`.
- Dashboard parses + stores capabilities; App gates the SkillsPanel \`onAcceptTrust\` prop on the flag. Older servers (without the field) → empty map → button hidden.

Closes #3272

## Test Plan

- [x] All new tests pass — 4 new auth-ok-handler tests (parse / default-empty / coerce-non-bool / array-rejected)
- [x] Existing tests unbroken — full dashboard suite (1382)
- [x] Dashboard typecheck + protocol build clean
- [x] Fail-closed semantics: missing capability ⇒ button hidden (matches #3270's back-compat path)

### CODE REVIEW

**Comment by blamechris** [comment]
## Code Review: PR #3273

### Summary

Closes the mixed-version-deployment gap from #3271's review. Server advertises a capability map in `auth_ok`; dashboard gates the SkillsPanel Accept button on `serverCapabilities.skillTrustAccept`. Older servers omit the field, the dashboard reads it as empty, the button hides — fail-closed.

### Strengths

- **Pattern reused, not invented.** `capabilities: z.record(z.string(), z.boolean()).optional()` already lives on `ServerProviderListSchema`. Mirroring the same shape for `auth_ok` keeps the protocol surface consistent and gives future flags a clear home.
- **Fail-closed parsing.** The handler coerces non-`true` values to `false`, treats arrays as missing, and defaults to `{}` on absent field. A malformed `capabilities: 1` from a buggy proxy can't accidentally enable a gate. Four direct tests lock all four edge cases.
- **Single reflective wiring.** App.tsx reads `s.serverCapabilities?.skillTrustAccept` once, passes either the action or `undefined`. The SkillsPanel's existing back-compat path (skip button when `onAcceptTrust` is undefined) handles the rest — no new branching in the panel.
- **Boolean strictness in the parser.** `serverCapabilities[k] = v === true` (not `!!v`) means `'true'`, `1`, `{}`, etc. become `false`. The tests pin this directly with `stringTrue: 'true'` and `numberOne: 1`.
- **Dist artifacts rebuilt.** Protocol consumers see the new field at the package boundary — same #3261 pattern.

### Issues Found

#### Critical (Must Fix)

_None._

#### Suggestions (Should Consider)

| File | Line | Suggestion | Rationale |
|------|------|------------|-----------|
| `packages/server/src/ws-history.js` | new `capabilities` block | Single hardcoded object today. Once a third capability lands, consider extracting to a top-level `SERVER_CAPABILITIES` constant alongside `PROTOCOL_VERSION`. | Defer until third flag — premature now. |
| `packages/dashboard/src/store/connection.ts` | `serverCapabilities: {}` initial | Could be a frozen object to prevent accidental mutation. Functionally identical given how the field is read. | Defer indefinitely. |

#### Nitpicks (Optional)

- The capability key naming convention is camelCase (`skillTrustAccept`) but the WS message it gates is snake_case (`skill_trust_accept`). A future contributor might name a key differently and break the convention. Worth a comment in `ws-history.js` declaring "keys are camelCase, value=true means the matching snake_case WS handler is implemented" if a third flag is added.

### Deferred Items (Follow-Up Issues)

_None._

### Architecture Notes

This is the right pattern for forward-evolving the protocol against mixed-version deployments. Adding a flag is a one-line server change + a one-line dashboard gate; the schema, parsing, and store wiring are all in place. Future feature-gated UI just adds a property to the capabilities object.

The fail-closed default is the correct choice for this surface: a button that does nothing is worse than a hidden feature, and "feature not advertised" is indistinguishable from "feature not supported" when the server omits the field.

### Verdict

- [x] **Approve** — Ready to merge

UX-quality follow-up landed cleanly. Pattern is reusable for any future feature with a similar mixed-version-deployment risk.

File: packages/dashboard/src/store/connection.ts [hunk 1]
```diff
@@ -345,6 +345,9 @@ export const useConnectionStore = create<ConnectionState>((set, get) => ({
   webFeatures: { available: false, remote: false, teleport: false },
   webTasks: [],
 
+  // #3272: capability map populated from auth_ok. Empty until connected.
+  serverCapabilities: {},
+
```
[the remark this answers is not in the document]
Comment (blamechris) [reply packages/dashboard/src/store/connection.ts:350]: **FIX**

Fixed in `268b5d77f`

**Change:** Clear `serverCapabilities` to `{}` in the disconnect path's `set()` call. A reconnect against a different (or older) server now sees a fail-closed empty map until `auth_ok` populates it from the new server.

```diff
  webFeatures: { available: false, remote: false, teleport: false },
  webTasks: [],
+ // #3272 review: clear advertised capabilities on disconnect so a
+ // reconnect against a different (or older) server can't have its
+ // UI gates left enabled by stale state. Empty map = fail-closed
+ // for any capability-gated affordance.
+ serverCapabilities: {},
  savedConnection: null,
```

Also added a test that pins the auth_ok-overwrites-stale-state contract: a reconnect with NO `capabilities` in the payload must clear prior capabilities, not merge.

File: packages/dashboard/src/App.tsx [hunk 2]
```diff
@@ -311,6 +311,10 @@ export function App() {
   // #3270: 'Accept new content' affordance — pairs with skill_trust_accept
   // server handler (#3235/#3269).
   const acceptSkillTrust = useConnectionStore(s => s.acceptSkillTrust)
+  // #3272: gate the Accept button on the server's advertised capability so
+  // older servers (which silently drop unknown WS messages) don't render a
+  // dead button. Treat missing flag as false — fail-closed.
+  const skillTrustAcceptSupported = useConnectionStore(s => !!s.serverCapabilities?.skillTrustAccept)
```
[the remark this answers is not in the document]
Comment (blamechris) [reply packages/dashboard/src/App.tsx]: **FIX**

Fixed in `268b5d77f`

**Change:** Added `connectionPhase === 'connected'` to the App's `skillTrustAcceptSupported` gate. Combined with the disconnect-clears-capabilities fix from comment 1, the button now only renders when both:
1. The socket is connected (so a click can actually send), AND
2. The current server has advertised `skillTrustAccept`.

```diff
- // #3272: gate the Accept button on the server's advertised capability so
- // older servers (which silently drop unknown WS messages) don't render a
- // dead button. Treat missing flag as false — fail-closed.
- const skillTrustAcceptSupported = useConnectionStore(s => !!s.serverCapabilities?.skillTrustAccept)
+ // #3272: gate the Accept button on (a) the server's advertised
+ // capability AND (b) an actually-connected socket. Without the
+ // connection check, capability state surviving from a previous
+ // connection could leave the button rendered while disconnected /
+ // reconnecting — clicks would then silently no-op. Treat missing
+ // flag as false — fail-closed.
+ const skillTrustAcceptSupported = useConnectionStore(s =>
+   s.connectionPhase === 'connected' && !!s.serverCapabilities?.skillTrustAccept,
+ )
```

**Comment by blamechris** [comment]
## Review Comments Addressed

| # | Comment | Outcome | Reference |
|---|---------|---------|-----------|
| 1 | serverCapabilities not cleared on disconnect — stale state across connections | FIX | `268b5d77f` |
| 2 | Accept button gate didn't check connection phase — could render while disconnected | FIX | `268b5d77f` |

**Total:** 2 comments addressed (1 commit). Both are facets of the same root issue. The fix is symmetric: clear capabilities on disconnect AND require a live connection in the UI gate. New test locks the auth_ok-overwrites-stale-state contract.
